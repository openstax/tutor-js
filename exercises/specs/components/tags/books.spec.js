import Renderer from 'react-test-renderer';
import { find } from 'lodash';
import Factory from '../../factories';
import Books from '../../../src/components/tags/books';


describe('Book tags component', function() {

    let props;

    beforeEach(() => {
        props = {
            exercise: Factory.exercise({ tags: ['book:stax-phys'] }),
        };
    });


    it('renders tags', () => {
        const books = Renderer.create(<Books {...props} />);
        expect(books.toJSON()).toMatchSnapshot();
        books.unmount();
    });

    it('can make choice', () => {
        const books = mount(<Books {...props} />);
        expect(find(props.exercise.tags, { value: 'stax-econ' })).toBeUndefined();
        books.find('select').simulate('change', { target: { value: 'stax-econ' } });
        expect(find(props.exercise.tags, { value: 'stax-econ' })).not.toBeUndefined();
        books.unmount();
    });

    it('can delete a tag', () => {
        const books = mount(<Books {...props} />);
        expect(props.exercise.tags).toHaveLength(1);
        books.find('Icon[type="trash"]').simulate('click');
        expect(props.exercise.tags).toHaveLength(0);
    });

    it('can add a new tag', () => {
        props.exercise.tags = [];
        const books = mount(<Books {...props} />);
        expect(books).not.toHaveRendered('select');
        books.find('.controls Icon[type="plus-circle"]').simulate('click');
        expect(books).toHaveRendered('select');
        books.unmount();
    });

});
