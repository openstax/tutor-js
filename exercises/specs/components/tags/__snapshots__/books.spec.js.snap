// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book tags component renders tags 1`] = `
.c0 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
  -webkit-transition: color 0.3s ease;
  transition: color 0.3s ease;
}

<div
  className="tag-type book has-single-tag"
>
  <div
    className="heading"
  >
    <span
      className="label"
    >
      Book
    </span>
    <div
      className="controls"
    >
      <button
        className="ox-icon-plus-circle btn btn-plain"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-plus-circle fa-w-16 c0 ox-icon ox-icon-plus-circle"
          data-icon="plus-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="tag"
  >
    <select
      className="form-control"
      onChange={[Function]}
      value="stax-phys"
    >
      <option
        value="stax-amfg"
      >
        Additive Manufacturing
      </option>
      <option
        value="stax-usgovt"
      >
        American Government
      </option>
      <option
        value="stax-anp"
      >
        Anatomy and Physiology
      </option>
      <option
        value="stax-anth"
      >
        Anthropology
      </option>
      <option
        value="stax-bio"
      >
        Biology
      </option>
      <option
        value="stax-apbio"
      >
        Biology for AP® Courses
      </option>
      <option
        value="stax-bca"
      >
        Business Computer Applications
      </option>
      <option
        value="stax-phys"
      >
        College Physics
      </option>
      <option
        value="stax-apphys"
      >
        College Physics for AP® Courses
      </option>
      <option
        value="stax-pyth"
      >
        Computer Programming with Python
      </option>
      <option
        value="stax-cs"
      >
        Computer Science
      </option>
      <option
        value="stax-cbio"
      >
        Concepts of Biology
      </option>
      <option
        value="stax-cmath"
      >
        Contemporary Math
      </option>
      <option
        value="stax-devpsy"
      >
        Developmental Psychology
      </option>
      <option
        value="stax-econ"
      >
        Economics
      </option>
      <option
        value="stax-engcomp"
      >
        English Composition
      </option>
      <option
        value="stax-eship"
      >
        Entrepreneurship
      </option>
      <option
        value="stax-fin"
      >
        Finance
      </option>
      <option
        value="stax-k12phys"
      >
        HS Physics
      </option>
      <option
        value="stax-infosys"
      >
        Information Systems
      </option>
      <option
        value="stax-macro"
      >
        Macro Economics
      </option>
      <option
        value="stax-micro"
      >
        Micro Economics
      </option>
      <option
        value="stax-orgchem"
      >
        Organic Chemistry
      </option>
      <option
        value="stax-phi"
      >
        Philosophy
      </option>
      <option
        value="stax-polisci"
      >
        Political Science
      </option>
      <option
        value="stax-mktg"
      >
        Principles of Marketing
      </option>
      <option
        value="stax-psy"
      >
        Psychology
      </option>
      <option
        value="stax-soc"
      >
        Sociology
      </option>
      <option
        value="stax-ushist"
      >
        U.S. History
      </option>
      <option
        value="stax-apush"
      >
        U.S. History for AP® Courses
      </option>
      <option
        value="stax-worldhist"
      >
        World History
      </option>
    </select>
    <span
      className="controls"
    >
      <button
        className="ox-icon-trash btn btn-plain"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 c0 ox-icon ox-icon-trash"
          data-icon="trash-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </span>
  </div>
</div>
`;
